PriceKataCalc/Program.cs

```diff
@@ -5,35 +5,52 @@ class priceCalc
     {
         public static void Main()
         {
-            product myProduct = new("ball", 20405, 15.1567,0.20,0.15);
-            Console.WriteLine($"Product price reported as {myProduct.price} before tax and {myProduct.calcPriceAfterTax()} after {myProduct.tax*100}% tax, And with {myProduct.discount*100}% discont its {myProduct.calcPriceAfterDiscount()}  ");
+            product myProduct = new("ball", 20405, 15.1567, 0.20, 0.15, true);
+            Console.WriteLine($"Product price reported as {myProduct.price} before tax and {myProduct.calcPriceAfterTax()} after {myProduct.tax * 100}% tax  ");
+            if (myProduct.haveDiscount)
+            {
+                Console.WriteLine($", And with {myProduct.discount * 100}% discont its {myProduct.calcPriceAfterDiscount()}");
+            }
+            else
+            {
+                Console.WriteLine("Program doesn’t show any discounted amount.");
 
+            }
         }
-       
+
     }
-    class product{
+    class product
+    {
         public String name;
         public int UPC;
         public double price;
         public double discount;
+        public bool haveDiscount;
         public double tax;
-        public product (String name,int UPC,double price, double tax,double discount) { 
-            this.name = name; 
+        public product(String name, int UPC, double price, double tax, double discount, bool haveDiscount)
```

Comment:
I don't think it is a good idea to have the the haveDiscount as part of the product. Since the product is the same but it some times I apply the discount and some times I don't .

Reply:
maybe creating another dictionary would solve the issue